ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -608,16 +610,40 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
         case MessageType_KtoU_NotifyDirectoryRenamed:
         case MessageType_KtoU_NotifyFileHardLinkCreated:
         {
+            string parentPath(request.path);
+            size_t lastDirSeparator = parentPath.find_last_of('/');
+            if (lastDirSeparator != string::npos)
+            {
+                parentPath = parentPath.substr(0, lastDirSeparator);
+                
+                char parentFullPath[PrjFSMaxPath];
+                CombinePaths(s_virtualizationRootFullPath.c_str(), parentPath.c_str(), parentFullPath);
+                
+                if (!IsBitSetInFileFlags(parentFullPath, FileFlags_IsInVirtualizationRoot))
+                {
+                    // TODO(Mac): Handle SetBitInFileFlags failures
+                    SetBitInFileFlags(parentFullPath, FileFlags_IsInVirtualizationRoot, true);
+                    
+                    HandleFileNotification(
```

Comment:
We need to walk up the tree and notify VFS for Git of all new folders it has not seen yet.